### PR TITLE
[imxrt] [driver] add usb device driver

### PR DESCRIPTION
**Pull request by tyustli**
## 拉取/合并请求描述：(PR description)

[
[imxrt] [driver] add usb device driver
verify platform:imxrt1052-atk-command
]

以下的内容不应该在提交PR时的message修改，修改下述message，PR会被直接关闭。请在提交PR后，浏览器查看PR并对以下检查项逐项check，没问题后逐条在页面上打钩。
The following content must not be changed in submitted PR message. Otherwise, the PR will be closed immediately. After submitted PR, please use web browser to visit PR, and check items one by one, and ticked them if no problem.

### 当前拉取/合并请求的状态 Intent for your PR

必须选择一项 Choose one (Mandatory):

- [ ] 本拉取/合并请求是一个草稿版本 This PR is for a code-review and is intended to get feedback
- [x] 本拉取/合并请求是一个成熟版本 This PR is mature, and ready to be integrated into the repo

### 代码质量 Code Quality：

我在这个拉取/合并请求中已经考虑了 As part of this pull request, I've considered the following:

- [x] 已经仔细查看过代码改动的对比 Already check the difference between PR and old code
- [x] 代码风格正确，包括缩进空格，命名及其他风格 Style guide is adhered to, including spacing, naming and other style
- [x] 没有垃圾代码，代码尽量精简，不包含`#if 0`代码，不包含已经被注释了的代码 All redundant code is removed and cleaned up
- [x] 所有变更均有原因及合理的，并且不会影响到其他软件组件代码或BSP All modifications are justified and not affect other components or BSP
- [x] 对难懂代码均提供对应的注释 I've commented appropriately where code is tricky
- [x] 本拉取/合并请求代码是高质量的 Code in this PR is of high quality

### CODE REVIEW

**Comment by BernardXiong** [comment]
配置应该有问题，gcc编译失败了

**Comment by tyustli** [comment]
@BernardXiong  熊大，看一下这个

File: bsp/imxrt/libraries/drivers/SConscript [hunk 1]
```diff
@@ -43,8 +45,24 @@ if GetDepend('BSP_USING_LCD'):
...
+if GetDepend('BSP_USING_USB_DEVICE'):
+    src += Glob('usb/phy/*.c')
+    CPPDEFINES += ['ENDIANNESS']
+
+if GetDepend('BSP_USING_USB_HOST'):
+    src += ['drv_usbh.c']
+    src += Glob('usb/host/*.c')
+
+if GetDepend('BSP_USING_USB_HOST'):
+    src += Glob('usb/phy/*.c')
```
Comment (lymzzyh) [inline bsp/imxrt/libraries/drivers/SConscript]: 如果同时开启会引入两次phy相关源码造成编译错误

File: bsp/imxrt/libraries/drivers/SConscript [hunk 2]
```diff
@@ -43,8 +45,24 @@ if GetDepend('BSP_USING_LCD'):
...
 
+if GetDepend('BSP_USING_USB_DEVICE'):
+    src += ['drv_usbd.c']
+    src += Glob('usb/device/*.c')
+
+if GetDepend('BSP_USING_USB_DEVICE'):
+    src += Glob('usb/phy/*.c')
+    CPPDEFINES += ['ENDIANNESS']
+
+if GetDepend('BSP_USING_USB_HOST'):
```
Comment (lymzzyh) [inline bsp/imxrt/libraries/drivers/SConscript]: 为什么device的驱动会引入host的相关代码，如果只是添加库文件的话似乎不需要再Sconscript中加入引用吧？

**Comment by BernardXiong** [comment]
"remote host config" ?? What's that?

**Comment by tyustli** [comment]
> "remote host config" ?? What's that?

sorry it is remove! because current driver not support host,so remove the host config

**Pull request merged**